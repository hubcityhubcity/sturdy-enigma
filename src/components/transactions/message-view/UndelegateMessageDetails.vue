<template>
  <div>
    <div class="tx__content__caption">
      <p>
        Undelegated
        <b>{{ coin.amount | atoms | prettyLong }}</b>
        <span>&nbsp;{{ coin.denom | viewDenom }}</span>
        <span v-if="transaction.liquidDate" class="tx-unbonding__time-diff"
          >&nbsp;{{ liquidDateCaption }}</span
        >
      </p>
    </div>
    <div class="tx__content__information">
      From&nbsp;
      <router-link :to="`/validators/${transaction.value.validator_address}`">
        {{
          transaction.value.validator_address | resolveValidatorName(validators)
        }}
      </router-link>
    </div>
  </div>
</template>

<script>
import moment from "moment"
import { atoms, viewDenom, prettyLong } from "scripts/num.js"
import { resolveValidatorName } from "src/filters"
import { getCoin } from "scripts/transaction-utils"

export default {
  name: `undelegate-message-details`,
  filters: {
    atoms,
    viewDenom,
    prettyLong,
    resolveValidatorName
  },
  props: {
    transaction: {
      type: Object,
      required: true
    },
    validators: {
      type: Object,
      required: true
    }
  },
  computed: {
    liquidDateCaption() {
      return `(liquid ${moment(this.transaction.liquidDate).fromNow()})`
    },
    coin() {
      return getCoin(this.transaction)
    }
  }
}
</script>

<style>
.tx__content__information,
.tx__content__information > * {
  display: flex;
  flex-direction: row;
}

.tx__content__information {
  font-size: 14px;
  color: var(--dim);
}

.tx__content__caption {
  line-height: 18px;
  font-size: 18px;
  color: var(--bright);
}
</style>
