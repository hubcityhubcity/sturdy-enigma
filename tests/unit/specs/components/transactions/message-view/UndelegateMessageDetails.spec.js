import { shallowMount } from "@vue/test-utils"
import UndelegateMessageDetails from "src/components/transactions/message-view/UndelegateMessageDetails"

describe(`UndelegateMessageDetails`, () => {
  let wrapper
  // Examples @ https://stargate.cosmos.network/txs?action=begin_unbonding
  const tx = {
    height: 2000000,
    value: {
      delegator_address: "cosmos16wgw3k7rspegl9khx79fuaa37unf4nfmxf7pyr",
      validator_address: "cosmosvaloper1sjllsnramtg3ewxqwwrwjxfgc4n4ef9u2lcnj0",
      amount: {
        denom: "uatom",
        amount: "100000"
      }
    },
    fee: {
      amount: "37",
      denom: "uatom"
    },
    hash: `ABCD1234`,
    group: `staking`,
    key: "keyhash",
    memo: "(Sent via Lunie)",
    timestamp: 123456789,
    liquidDate: null,
    type: `cosmos-sdk/MsgUndelegate`
  }

  it(`renders a undelegate transaction message`, () => {
    wrapper = shallowMount(UndelegateMessageDetails, {
      propsData: {
        transaction: tx,
        validators: {}
      },
      stubs: [`router-link`]
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
