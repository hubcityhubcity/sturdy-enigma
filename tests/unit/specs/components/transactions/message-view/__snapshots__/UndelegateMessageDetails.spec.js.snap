// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UndelegateMessageDetails renders a undelegate transaction message 1`] = `
<div
  class="tx__content"
>
  <transactionicon-stub
    transactiongroup="staking"
    transactiontype="Unstaked"
  />
   
  <div
    class="tx__content__left"
  >
    <h3>
      Unstaked
    </h3>
     
    <span>
      from 
    </span>
     
    <router-link-stub
      to="/staking/validators/cosmosvaloper1sjllsnramtg3ewxqwwrwjxfgc4n4ef9u2lcnj0"
    >
      <!---->
      
      cosmosvaloper…cnj0
    
    </router-link-stub>
  </div>
   
  <div
    class="tx__content__right"
  >
    <p
      class="amount"
    >
      
      0.1 ATOM
    
    </p>
  </div>
</div>
`;
